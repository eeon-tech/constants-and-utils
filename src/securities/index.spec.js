const _ = require('lodash')
const securities = require('./index')

describe('securities', () => {
  describe('scopes', () => {
    it('should stay consistent', () => {
      expect(securities.scopes).toMatchSnapshot()
    })
  })

  describe('getMarketIndicesScopeName', () => {
    it('should return the scope name', () => {
      expect(securities.getMarketIndicesScopeName()).toBe(
        _.get(securities, ['scopes', 'MARKET_INDICES', 'name'])
      )
    })
  })

  describe('getSectorIndicesScopeName', () => {
    it('should return the scope name', () => {
      expect(securities.getSectorIndicesScopeName()).toBe(
        _.get(securities, ['scopes', 'SECTOR_INDICES', 'name'])
      )
    })
  })
})
