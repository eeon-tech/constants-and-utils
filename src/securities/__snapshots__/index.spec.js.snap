// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`securities scopes should stay consistent 1`] = `
Object {
  "MARKET_INDICES": Object {
    "name": "marketIndices",
    "options": Object {
      "where": Object {
        "symbol": Array [
          "DIA",
          "IWM",
          "QQQ",
          "SPY",
        ],
      },
    },
  },
  "SECTOR_INDICES": Object {
    "name": "sectorIndices",
    "options": Object {
      "where": Object {
        "symbol": Array [
          "XLB",
          "XLC",
          "XLE",
          "XLF",
          "XLI",
          "XLK",
          "XLP",
          "XLRE",
          "XLU",
          "XLV",
          "XLY",
        ],
      },
    },
  },
}
`;
